### ISSUE TITLE
download DHS model datasets with get_datasets()

### ISSUE THREAD

**Issue by jeffeaton** [OJWatson/rdhs#33]
Can we make `get_datasets()` work for the [DHS model datasets](https://dhsprogram.com/data/download-model-datasets.cfm)? i.e.

```
get_datasets("ZZBR62FL.ZIP")
get_datasets("ZZIR62FL.ZIP")
get_datasets("ZZMR61FL.ZIP")
get_datasets("ZZAR61FL.ZIP")
...
```

It would probably be a bit hacky under the hood, but would be convenient for demonstrating integration of rdhs with other packages. The URLs are:

```
https://dhsprogram.com/customcf/legacy/data/sample_download_dataset.cfm?Filename=ZZBR62FL.ZIP&Tp=1&Ctry_Code=zz&survey_id=0&doctype=dhs
https://dhsprogram.com/customcf/legacy/data/sample_download_dataset.cfm?Filename=ZZIR62FL.ZIP&Tp=1&Ctry_Code=zz&survey_id=0&doctype=dhs
https://dhsprogram.com/customcf/legacy/data/sample_download_dataset.cfm?Filename=ZZMR61FL.ZIP&Tp=1&Ctry_Code=zz&survey_id=0&doctype=dhs
https://dhsprogram.com/customcf/legacy/data/sample_download_dataset.cfm?Filename=ZZAR61FL.ZIP&Tp=4&Ctry_Code=zz&survey_id=0&doctype=hiv
```
etc.

**Comment by jeffeaton** [comment]
Yeah, that's what I had in mind. Catch them for now just to download the dataset using `rdhs::get_datasets("ZZIR62FL.ZIP")`. When Trevor to get them into the API all the better, but not essential.


Use case is that I am doing some work on the demographic analysis package to get some code to colleagues and need examples. Here's what I'm currently put in the tests and examples:


```

irfl_zip <- tempfile()
download.file("https://dhsprogram.com/customcf/legacy/data/sample_download_dataset.cfm?Filename=ZZIR62FL.ZIP&Tp=1&Ctry_Code=zz&survey_id=0&doctype=dhs", irfl_zip)

ir <- rdhs::read_dhs_flat(irfl_zip)
```

Using `rdhs::get_datasets("ZZIR62FL.ZIP")` would be a bit more concise, and put it in the `rdhs` cache so that I don't re-download from their server everytime.

________________________________

**Comment by OJWatson** [comment]
Okay i'll get round to this some point this week. Lemme know if it's needed earlier than some point. My current plan is to actually create the data.frame of what the model datasets would look like, using the individual url links from the model datasets page etc, and then row bind that dataframe when needed to calls of available_datasets and dhs_datasets. That way it fits in correctly, and the user still goes through the correct code.

**Comment by OJWatson** [comment]
something like this: https://github.com/OJWatson/rdhs/commit/96c29e13a5b5a1ff50610e2fe9f6a8bc271439d8

**Comment by OJWatson** [comment]
So this is coming through in the development pipeline, and is just checking at the mo. but looks like this now:

```
dat <- get_datasets(dataset_filenames = "zzar61.zip",
                      download_option = "zip")
```

and that works fine. Will close when it's merged into master along with v0.5

**Comment by OJWatson** [comment]
This is now handled as well in v0.5.1
